src/test/java/teammates/test/cases/webapi/ArchiveCourseActionTest.java

```diff
@@ -28,71 +29,59 @@ protected String getRequestMethod() {
     protected void testExecute() throws Exception {
         InstructorAttributes instructor1OfCourse1 = typicalBundle.instructors.get("instructor1OfCourse1");
         String instructorId = instructor1OfCourse1.googleId;
-        String courseId = instructor1OfCourse1.courseId;
 
         loginAsInstructor(instructorId);
 
         ______TS("Not enough parameters");
         verifyHttpParameterFailure();
-        verifyHttpParameterFailure(Const.ParamsNames.COURSE_ID, courseId);
-        verifyHttpParameterFailure(Const.ParamsNames.COURSE_ARCHIVE_STATUS, "true");
 
         ______TS("Typical case: archive a course");
 
         String[] submissionParams = new String[] {
                 Const.ParamsNames.COURSE_ID, instructor1OfCourse1.courseId,
-                Const.ParamsNames.COURSE_ARCHIVE_STATUS, "true",
         };
 
-        ArchiveCourseAction archiveAction = getAction(submissionParams);
+        CourseArchiveRequest courseArchiveRequest = new CourseArchiveRequest();
+        courseArchiveRequest.setArchiveStatus("true");
+
+        ArchiveCourseAction archiveAction = getAction(courseArchiveRequest, submissionParams);
         JsonResult result = getJsonResult(archiveAction);
 
         assertEquals(HttpStatus.SC_OK, result.getStatusCode());
```

Comment:
Not your fault. I think we should add a verification (i.e. the course is indeed archived) here.

There are multiple places where the verification lacks. If you have time, please help to fix it.